### DISCUSSION TITLE
Extracting values as environment variables

### DISCUSSION THREAD

**Discussion by andrewmiskell** [OpenIDC/mod_auth_openidc#634] [Q&A]
I'm trying to figure out how I'd be able to extract additional values from the JWT data in order to pass additional information to an application.

I see how to specify the value for REMOTE_USER using OIDCRemoteUserClaim, but I'm like to create two additional variables for real name (using the name value in the JWT) and one for unique user (using the sub value in the JWT as email would be the REMOTE_USER variable).

I've read through the conf file but still a little unsure on exactly how this would work, if possible at all.

**Comment by zandbelt** [comment]
you can use `SetEnfIfExpr` https://httpd.apache.org/docs/2.4/mod/mod_setenvif.html#setenvifexpr use `OIDC_CLAIM_*` environment variables set by mod_auth_openidc in an expression to set another environment variable

**Reply by OlivierBOEL** [reply]
Could you please provide an example?

**Reply by zandbelt** [reply]
it turns out one actually needs mod_rewrite to set an environment variable that refers to another variable:
```
   RewriteEngine On
   RewriteRule .* - [E=MY_NAME:%{HTTP:OIDC_CLAIM_name}]
```
